packages/concerto-core/lib/introspect/classdeclaration.js

```diff
@@ -190,21 +190,6 @@ class ClassDeclaration extends Declaration {
     validate() {
         super.validate();
 
-        const declarations = this.getModelFile().getAllDeclarations();
```

Comment:
There is nothing in this removed code that references the particular declaration (`this`,) and it already calls the model file. So we can just move this logic to model file and remove unnecessary O(n^2) loop.